### ISSUE TITLE
Tor bridges logs appears on the debug script

### ISSUE THREAD

**Issue by nyxnor** [getumbrel/umbrel#727]
https://github.com/getumbrel/umbrel/blob/b5bbef195d7c3e9cf395788d60edce8e67c4f834/scripts/debug#L137

I've read the security policy.

Even with notice logs, bridges IP and names are shown on the logs. It is not possible to hide from the Tor logs even using notice.
Bridges information doesnt only appear when bootstrapping, but from time to time when receiving new relevant data and when cannot reach them. The only moment with bridges when their info does not appear is when there are no running bridges.
```
new bridge descriptor 'NAME' (fresh): FINGERPRINT
Bridge 'NAME' has both an IPv4 and an IPv6 address.  Will prefer using its IPv4 address (IP) MOREINFOHERE*
```
This is an attack vector and should be removed. This is not okay to use with `--upload`. The enemy wont have the entire network to search for the specific user, but only 3 (bridges). This is one of the reasons why your bridges should not be shown.

I don't know the proper solution to still present Tor logs and hide bridge, cause the user might want to see if there is running bridges on his logs.
`sudo systemctl is-active tor@default`
This is not a solution, because if there is no running bridges, the status will still be active. A more complex solution than this would be to hide every log mentioning the word `bridge` except `no running bridges`. 

I won't accept this is not relevant because only few people use Tor bridges. If they need to use it, they can't be compromised. The bridges you use should never be shared.

**Comment by AaronDewes** [comment]
> I won't accept this is not relevant because only few people use Tor bridges. If they need to use it, they can't be compromised. The bridges you use should never be shared.

While this is relevant, it's not a priority right now. The dashboard will get a new debug feature, where you can review everything before uploading, and you can remove data yourself and use a different service if you don't want something exposed. So it might get added in the future, but not now.

**Comment by nyxnor** [comment]
>use a different service if you don't want something exposed

That is better than using only umbrel's service, excellent. The bad part is that the bridges info will still be shared if the user does not exclude or you modify the script.

>and you can remove data yourself

Do you think the user wont forget to delete in the process? 
Please consider this:
`sudo systemctl status tor@default | grep -v 'Bridge\|bridge'`

This solves as far as I know.
Because if there is no running bridges, it will be in the plural form with `s`. Also, if it there is wrong bridge information in the torrc, it won't bootstrap.

**Comment by AaronDewes** [comment]
`sudo systemctl status tor@default | grep -v 'Bridge\|bridge'` The grep would have to be added after the `docker-compose logs --tail=30 tor`, because tor runs in docker. Please try commands before sharing them :wink:, but we'll add this.

@louneskmt Can you also add this for your debug UI?

> That is better than using only umbrel's service, excellent. 

Why is that better? It sounds like you find it generally better. For me as someone who provides support at least, our service is better because it only shows important information, is ad-free and has separate tabs for the logs and dmesg.

**Comment by nyxnor** [comment]
@AaronDewes I dont use docker, you are right, try just the grep part as you mentioned. Sorry for that.
>Why is that better?

Freedom of choice. For the owners of the domain, is all good, all logs automatically being uploaded there. But this does give the user the choice, as he would need to modify the script.

**Comment by AaronDewes** [comment]
> Freedom of choice. For the owners of the domain, is all good, all logs automatically being uploaded there. But this doesn't give the user the choice, as he would need to modify the script.

Yeah, that's why we don't upload by default, but only with a specific flag. So automated uploads as a flag for the script or something isn't possible, because paste servers aren't standardized. But this server is the best option for uploads, because it's clean, ad-free, doesn't get requests blocked by Cloudfare DDoS protection (an issue we had with hastebin), parses bash colors properly and can get adjusted to be better for Umbrel logs. But that's just my opinion, and I'm the owner and developer of the server. There are many bash functions to upload though, so you can do stuff like

```
# Function to upload
upload() { echo "Code here" }

./scripts/debug | some filter here | upload
```

But the bridge removal is definitely something we'll add soon.

**Comment by nyxnor** [comment]
Thank you. Good explanation above. Always good to be corrected. Just hope this soon is really soon.

**Comment by louneskmt** [comment]
Hey @nyxnor! Thanks for your concern and time.

> Edit: Consider onion service for the users to upload.

I'm currently adding an UI to get the logs and some debug informations from the dashboard. The upload now uses Tor, and @AaronDewes will soon add an onion URL for the server. See https://github.com/getumbrel/umbrel-manager/pull/66/commits/3232fceac71e66a1d082eeb42745519ffa899f1f.

I will make a PR to remove any sensitive data about Tor Bridges from the logs. Btw, Umbrel doesn't use Bridges by default (At least this is not configured. Can bridges be chosen and used without any configuration?).

Do you think this would be sufficient (or the `sed`equivalent)?
```
grep -v 'Bridge\|bridge'
```

**Comment by nyxnor** [comment]
@louneskmt 

>Btw, Umbrel doesn't use Bridges by default (At least this is not configured. Can bridges be chosen and used without any configuration?).

Im not sure if this was for me. But I will answer to help.
Normal bridges can be used, but normal bridges does not hide the user is using Tor, for this, he would need to use a transport such as obfs4 or meek. This package does not work on arm (32 or 64) from the repo, needs to be compiled from source.
Bridges need to be included manually, there is no mean to acquire this with terminal commands, as there is captchas to solve.
Also need to configure `torrc` to use them.

>Do you think this would be sufficient (or the sed equivalent)?

Again, not sure if this is for me. Sed also. This is a workaround, I've done some tests, but I have a small sample. If there is a more ideal way to do this, I dont know how, as Tor does not hide these logs.

**Comment by louneskmt** [comment]
Thanks for your answer. The questions were indeed for you 😁

As Umbrel doesn't use Bridges for now (this is not configured and the package obfs4 is not installed), do you think it is relevant to hide some logs? 

But for sure, we can always add this in case a power user adds them manually.

**Comment by nyxnor** [comment]
Yes, because in the case an user has Tor connections blocked, if he searches how to do it, he will try to use bridges. In this case, should be hidden as default, case even if the user does not use bridges, it wont cause any problems, and if he is using, will be more safe for him.
pkg name is obfs4proxy, forgot to mention.

**Comment by nyxnor** [comment]
How is this issue still open? I thought it was gonna be fixed that day....
https://www.reddit.com/r/getumbrel/comments/nkl2f5/umbrel_in_china/
A reddit user was instructed to use Tor Bridges, if he does, will leak to vercel app.
Pls fix.
If still don't think it is a major issue, go to #tor irc chat and ask there about sharing personal bridges to a server.
(I won't PR cause I don't use docker and because of licensing)

**Comment by lukechilds** [comment]
Thanks for reporting this @nyxor.

>A reddit user was instructed to use Tor Bridges

Do you have a link to that? Bridges aren't mentioned in the thread you linked. In fact, bridges aren't supported in Umbrel in the first place.

>if he does, will leak to vercel app.

Even if the user manually modified their install to support bridges (which would be stripped out on every OTA update), the data would only be leaked to Vercel if they SSH into their Umbrel and manually upload the logs to Vercel. Neither of these things are recommended. We've built troubleshooting capabilities into the UI that provide the user with a log file they can download and privately share with us.

**Comment by nyxnor** [comment]
Why did you close? This is not fixed.
Bridges are supported if you have Tor.

**Comment by nyxnor** [comment]
>even if the user manually modified their install to support bridges 

Not needed, they just need to change torrc and sighup tor, as explained above

**Comment by lukechilds** [comment]
>Not needed, they just need to change torrc and sighup tor, as explained above

Changing the configuration is modifying the installation.

>Why did you close? This is not fixed.

If users go changing random values in configuration files they can do all sorts of dangerous things.

The whole point of Umbrel is that it's already configured in a safe way. We can't prevent 1000s of security footguns if users start modifying config files themselves.

**Comment by nyxnor** [comment]
>If users go changing random values in configuration files they can do all sorts of dangerous things.

No random values, all acquired from a trusted DB from Tor Project https://bridges.torproject.org/

>If users go changing random values in configuration files they can do all sorts of dangerous things.

Possible, instruct them how to use Tor Bridges safely.

>The whole point of Umbrel is that it's already configured in a safe way

You know this is not right, Umbrel is not secure by default.

>We can't prevent 1000s of security footguns if users start modifying config files themselves.

Agree with this, they need proper instruction. Filter the logs or make some writing in the repo about security with Tor bridges.

**Comment by nyxnor** [comment]
>Do you have a link to that? Bridges aren't mentioned in the thread you linked. 

Yes, no proper instruction there, just recommending vanilla Tor and VPN, comment there for them to use Tor Bridges please for their safety, by using normal relays, their connection is blocked and possibly will be hunt down, you can turn the table.

**Comment by lukechilds** [comment]
So to our knowledge no one has ever been affected by this.

You want us to drop what we're doing to make code changes to protect some hypothetical person in some hypothetical scenario from doing something that goes against multiple recommendations we make.

Umbrel doesn't support bridges. I'd love to add support for them in the future. If/when that happens we'll add it as an option in the UI that will be enabled safely and we'll be sure to strip any sensitive data that may add to log files.

**Comment by nyxnor** [comment]
>So to our knowledge no one has ever been affected by this.

You can't know every user configuration, fair, but you can guide them to have more security

>You want us to drop what we're doing to make code changes to protect some hypothetical person in some hypothetical scenario from doing something that goes against multiple recommendations we make.

Censorship is not hypothetical, it is just not in your whereabouts https://metrics.torproject.org/userstats-bridge-country.html

>from doing something that goes against multiple recommendations we make.

I recommend them using Tor Bridges if their Tor connection is blocked, contacting TPO If in need https://www.torproject.org/contact/. You recommend againsta= changing configuration file even if it could save their lives?

>Umbrel doesn't support bridges

It does, it is just not one click to use it.

>f/when that happens we'll add it as an option in the UI that will be enabled safely and we'll be sure to strip any sensitive data that may add to log files.

Please do.
